patterns/behavioural/sealed.md

```diff
@@ -0,0 +1,79 @@
+# Sealed Trait
+
+## Description
+
+In Rust, a sealed trait is a trait that can only be implemented within the same
+crate where it is defined. It is achieved by making a public trait that depends
+on a [supertrait](https://doc.rust-lang.org/rust-by-example/trait/supertraits.html)
+that is private or only public on the local crate. This restricts the ability
+to implement the trait from outside the crate and provides a form of interface
+segregation.
+
+## Motivation
+
+Sealed traits can be used to create a set of behaviors that that allow future
+addition of methods without breaking the API.
+
+The sealed trait pattern helps to separate the public interface from the
+implementation details. This makes it easier to maintain and evolve the
+implementation over time without breaking existing code. It also provides a
+level of abstraction that allows users to interact with the library or module at
+a high level, without needing to know the implementation details.
+
+Because users of this crate can't implement this trait directly, is possible to
+add methods to it, without break existing code using this create. Only
+implementations of this trait will need updated, what is assured by the
+sealed trait to only happen locally.
+
+## Example
+
+The standard library makes use of a sealed trait, one example is the
+`OsStrExt` trait for
+[unix](https://doc.rust-lang.org/std/os/unix/ffi/trait.OsStrExt.html),
+[windows](https://doc.rust-lang.org/std/os/windows/ffi/trait.OsStrExt.html) and
+[wasi](https://doc.rust-lang.org/std/os/wasi/ffi/trait.OsStrExt.html).
```

Comment:
Can we have more examples, IIRC serde does this too.

Reply:
I could not find a good example from serde. They all seem too complex and obscure.

Reply:
I added a simpler example with a more defined objective.